DROD/RoomWidget.cpp

```diff
@@ -4734,6 +4747,9 @@ void CRoomWidget::Paint(
 	if (!bPlayerIsAlive) // To ensure everything draws properly during death animation everything must be made dirty
 		this->bAllDirty = true;
 
+	if ((this->bAllDirty || bMoveMade) && bPlayerIsAlive)
```

Comment:
I'm wondering whether the performance issue has to do with regenerating them on "bAllDirty". My intuition is that we don't need to rebuild all the effects each time the room is fully refreshed -- only when the turn changes.

The other situation I can think of is when there are no EMOVEORDERHINT effects in the list, but should be. Otherwise, why rebuild them?

Reply:
The delays are happening between turns being processed, so regenerating on refresh might not be the issue.

A little bit of profiling suggests the heaviest parts are in SDL_tff, but I couldn't get symbols loading for a closer look. Unfortunatey I don't think there's a good way to cut down on the "drawing text" part of this.

Reply:
Thanks for checking. One idea would be to maintain a singleton map caching pre-rendered surfaces, i.e., <turn number, Surface*>. Reuse the surfaces based on the turn number. This should resolve that issue and help avoid a lot of memory address space fragmentation as these surfaces are continuously being deleted and rebuilt.

Reply:
The caching solution worked out fairly well. I doubt most rooms will get anywhere near a thousand monsters, but it gets a little sluggish around the 300 to 400 mark.

Technically, you can get movement orders over 1216 by using the `Invisible includes move order` on invisible inspectable characters (well, you can now I've bundled in a fix to it with this PR). This is very unlikely in my opinion.

Reply:
Great work! Thank you for implementing the cache. I like how you designed the static clean-up.

Your reply gives me one more thought. Brainstorming, how about we impose an upper bound on the number of possible cached entries? Specifically, what I'm thinking is either:
1. With a ton of monsters in the room, tracking movement order for every one is probably not useful. How about for all entries with a value >= 1000, show the same (cached) symbol, like "+"?
2. Simply don't show an effect for any monster w/ order above a magic number, e.g., >= 1000.

Reply:
After thinking about it, I decided to have all values >=1000 be disaplayed as `>999`. It's not quite arbitary, since a five character string won't fit on a tile. The exact value is still available through right-click, and if a player needs that information it's perhaps arguable that something has gone wrong somewhere in terms of room design.